lib/ansible/modules/system/known_hosts.py

```diff
@@ -121,6 +121,11 @@ def enforce_state(module, params):
     if key and not key.endswith('\n'):
         key += '\n'
 
+    # Preserve file permission for path since they are lost due to issue in
+    # ssh-keygen command
+    # Remove this once ssh-keygen behaves normally
+    prev_mode = os.stat(path)
```

Comment:
i would add if os.path.exists check, then default to a 'safe/secure' default on else